src/OpenLoco/Map/Tile.h

```diff
@@ -182,6 +176,12 @@ namespace OpenLoco::Map
             _type |= state ? 0x40 : 0;
         }
         void createWave(int16_t x, int16_t y, int animationIndex);
+        bool hasHighTypeFlag() const { return _type & 0x80; }
+        void setHighTypeFlag(bool state)
+        {
+            _type &= ~0x80;
+            _type |= state ? 0x80 : 0;
+        }
```

Comment:
Why did you remove these members from `TileElementBase` to `SurfaceElement`?

Reply:
They weren't generic to all tile elements so i moved them to the specific instance.